src/main.ts

```diff
@@ -23,13 +24,24 @@ function addCustomPindropFunctionality(map: L.Map): void {
   });
 } 
 
+let weatherSubscription$: Subscription | null = null;
+
 async function setWeather(city: CityInterface): Promise<void> {
-  toggleWeatherContainer();
   try {
-      const [currentWeatherData, hourlyWeatherData] = await getWeatherData(city);
-      populateCurrentWeather(currentWeatherData, city);
-      populateHourlyWeather(hourlyWeatherData, currentWeatherData);
-      toggleWeatherContainer();
+    if (weatherSubscription$) {
+        weatherSubscription$.unsubscribe();
+    }
+    weatherSubscription$ = getWeatherObservable(city).subscribe({
```

Comment:
This is the only real issue I have with your code. It's not needed inside a promise, just subscribe and handle the response as you have done.

Reply:
I removed it